pkg/compactor/block_upload.go

```diff
@@ -644,7 +662,7 @@ func (c *MultitenantCompactor) GetBlockUploadStateHandler(w http.ResponseWriter,
 
 	s, _, v, err := c.getBlockUploadState(r.Context(), userBkt, blockID)
 	if err != nil {
-		writeBlockUploadError(err, "get block state", "", log.With(util_log.WithContext(r.Context(), c.logger), "block", blockID), w)
+		writeBlockUploadError(err, "", logger, w)
```

Comment:
I'd try to make sure that each call to `writeBlockUploadError` has some unique string that makes it easy to lookup the location in the code where an error was produced. 
In many other places this unique string is part of the `err` variable, so it's fine to pass `""` as second argument, but in this case here the `err` variable has no string which is quick to look up.

Reply:
same on line `95`.